From e65ff59aac8de420380271143f6d6204940bdc73 Mon Sep 17 00:00:00 2001
From: Casey Schaufler <casey@schaufler-ca.com>
Date: Thu, 10 May 2018 14:25:23 -0700
Subject: [PATCH 1518/1548] LSM: Infrastructure management of the superblock
 security blob

Move management of the superblock->sb_security blob out
of the individual security modules and into the security
infrastructure. Instead of allocating the blobs from within
the modules the modules tell the infrastructure how much
space is required, and the space is allocated there.

Signed-off-by: Casey Schaufler <casey@schaufler-ca.com>
---
 security/smack/smack.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/security/smack/smack.h b/security/smack/smack.h
index 87ced0fc1a19..1f0856a0cecf 100644
--- a/security/smack/smack.h
+++ b/security/smack/smack.h
@@ -433,6 +433,12 @@ static inline struct smack_known **smack_key(const struct key *key)
 }
 #endif /* CONFIG_KEYS */
 
+static inline struct superblock_smack *smack_superblock(
+					const struct super_block *superblock)
+{
+	return superblock->s_security;
+}
+
 /*
  * Is the directory transmuting?
  */
-- 
2.20.1

